app/models/banner.rb

```diff
@@ -2,32 +2,37 @@ class Banner < ActiveRecord::Base
   include HasPublicationStates
 
   has_one :link, as: :linkable, dependent: :destroy
+  has_many :pages, dependent: :nullify, inverse_of: :banner
 
   accepts_nested_attributes_for :link
 
-  delegate :url, :text, to: :link, prefix: true
+  delegate :url, :text, to: :link, prefix: true, allow_nil: true
 
-  validates :key, uniqueness: true, allow_nil: true
+  before_save(if: :became_default?) do |banner|
```

Comment:
Unused block argument - `banner`. You can omit the argument if you don't care about it.